Code/MinimalLib/docker/Dockerfile

```diff
@@ -1,7 +1,39 @@
+# Example usage of this Dockerfile:
+# (the build-arg arguments are all optional)
+#
+# 1. cd to Code/MinimalLib/docker
+# cd Code/MinimalLib/docker
+#
+# 2. build the JS and WASM libraries
+#    (the build-arg arguments are all optional)
+# docker build -t rdkit-minimallib --network=host \
+#  --build-arg "RDK_MINIMAL_LIB_SUPPORT_LEGACY_BROWSERS=ON" \
+#  --build-arg "RDKIT_GIT_URL=https://github.com/myfork/rdkit.git" \
+#  --build-arg "RDKIT_BRANCH=mybranch"
+#
+# 3. create a temporary container and copy built libraries
+#    from the container to your local filesystem, then destroy
+#    the temporary container
+# docker create --name=rdkit-minimallib-container rdkit-minimallib:latest --entrypoint /
+# docker cp rdkit-minimallib-container:/RDKit_minimal.js ../demo
+# docker cp rdkit-minimallib-container:/RDKit_minimal.wasm ../demo
+# docker cp rdkit-minimallib-container:/RDKit_minimal_legacy.js ../demo
+# docker rm rdkit-minimallib-container
+
+
+ARG RDK_MINIMAL_LIB_SUPPORT_LEGACY_BROWSERS="OFF"
+ARG RDKIT_GIT_URL="https://github.com/rdkit/rdkit.git"
+ARG RDKIT_BRANCH="master"
+
 FROM debian:buster as build-stage
+ARG RDK_MINIMAL_LIB_SUPPORT_LEGACY_BROWSERS
+ARG RDKIT_GIT_URL
+ARG RDKIT_BRANCH
```

Comment:
Sorry for my docker ignorance, but why do you have the `ARG` lines a second time?

Reply:
`ARG` declarations are local to each stage as identified by the `FROM` directive.
Here I wanted a global default at the top of the file for the `RDK_MINIMAL_LIB_SUPPORT_LEGACY_BROWSERS`, `RDKIT_GIT_URL`, `RDKIT_BRANCH` parameters in case they are not defined on the command line.
If you do not redeclare the `ARG`s in each stage you want to use them, they won't be defined.

Simple example; this works as expected:
```
$ cat Dockerfile
ARG RDKIT_GIT_URL="https://github.com/rdkit/rdkit.git"

FROM debian:buster as build-stage
ARG RDKIT_GIT_URL

RUN echo RDKIT_GIT_URL ${RDKIT_GIT_URL}

$ docker build --no-cache -t rdkit-minimallib --network=host --build-arg "RDKIT_GIT_URL=https://github.com/ptosco/rdkit.git" .
Sending build context to Docker daemon    107kB
Step 1/4 : ARG RDKIT_GIT_URL="https://github.com/rdkit/rdkit.git"
Step 2/4 : FROM debian:buster as build-stage
 ---> d4729731066b
Step 3/4 : ARG RDKIT_GIT_URL
 ---> Running in 879eae5baea0
Removing intermediate container 879eae5baea0
 ---> 31f10c169169
Step 4/4 : RUN echo RDKIT_GIT_URL ${RDKIT_GIT_URL}
 ---> Running in 09826c13acad
RDKIT_GIT_URL https://github.com/ptosco/rdkit.git
Removing intermediate container 09826c13acad
 ---> ea52dc9ee55c
Successfully built ea52dc9ee55c
Successfully tagged rdkit-minimallib:latest
```
If I comment the second `ARG` directive, `RDKIT_GIT_URL` is now undefined:
```
$ cat Dockerfile
ARG RDKIT_GIT_URL="https://github.com/rdkit/rdkit.git"

FROM debian:buster as build-stage
#ARG RDKIT_GIT_URL

RUN echo RDKIT_GIT_URL ${RDKIT_GIT_URL}

$ docker build --no-cache -t rdkit-minimallib --network=host --build-arg "RDKIT_GIT_URL=https://github.com/ptosco/rdkit.git" .
Sending build context to Docker daemon    107kB
Step 1/3 : ARG RDKIT_GIT_URL="https://github.com/rdkit/rdkit.git"
Step 2/3 : FROM debian:buster as build-stage
 ---> d4729731066b
Step 3/3 : RUN echo RDKIT_GIT_URL ${RDKIT_GIT_URL}
 ---> Running in 4ff02ecd5467
RDKIT_GIT_URL
Removing intermediate container 4ff02ecd5467
 ---> 4f4212b320be
Successfully built 4f4212b320be
Successfully tagged rdkit-minimallib:latest
```
```

Reply:
thanks for the explanation and example. That's sooooo weird.